spec/facility_spec.rb

```diff
@@ -23,4 +34,105 @@
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+  end
+  
+# GETTING A DRIVER'S LICENSE
+  describe 'getting a drivers license' do
+    it 'takes the written test' do
+      expect(@registrant_1.license_data).to eq({:written => false, :license => false, :renewed => false})
+      expect(@registrant_1.permit?).to be true 
+      @facility_1.add_service("Written Test")
+      @facility_1.administer_written_test?(@registrant_1)
+      expect(@facility_1.administer_written_test?(@registrant_1)).to be true
+      expect(@registrant_1.license_data).to eq({:written => true, :license => false, :renewed => false})
+      
+      expect(@registrant_2.license_data).to eq({:written => false, :license => false, :renewed => false})
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to be false
+      expect(@facility_1.administer_written_test?(@registrant_2)).to be false
+      @registrant_2.earn_permit
+      expect(@facility_1.administer_written_test?(@registrant_2)).to be true
+      expect(@registrant_2.license_data).to eq({:written => true, :license => false, :renewed => false})
+
+      expect(@registrant_3.license_data).to eq({:written => false, :license => false, :renewed => false})
+      expect(@registrant_3.age).to eq(15)
+      @registrant_3.earn_permit
+      expect(@registrant_3.permit?).to be true
+      expect(@facility_1.administer_written_test?(@registrant_3)).to be false
+    end
+
+    it 'takes the road test' do
+      expect(@facility_1.administer_road_test?(@registrant_3)).to be false
+      @registrant_3.earn_permit
+      expect(@facility_1.administer_road_test?(@registrant_3)).to be false
+      expect(@registrant_3.license_data).to eq({:written => false, :license => false, :renewed => false})
+
+      expect(@facility_1.administer_road_test?(@registrant_1)).to be false
+      @facility_1.add_service("Written Test")
+      @facility_1.add_service('Road Test')
+      expect(@facility_1.services).to eq(["Written Test", "Road Test"])
+
+      expect(@facility_1.administer_road_test?(@registrant_1)).to be true
+      expect(@registrant_1.license_data).to eq({:written => true, :license => true, :renewed => false})
+    end
+
+    it 'renews their license' do
+      @facility_1.add_service("Written Test")
+      @facility_1.add_service('Road Test')
+      
+      @facility_1.administer_written_test?(@registrant_1)
+      @facility_1.administer_road_test?(@registrant_1)
+      expect(@registrant_1.license_data).to eq({:written => true, :license => true, :renewed => false})
+    
+      @facility_1.add_service('Renew License')
+      expect(@facility_1.services).to eq ["Written Test", "Road Test", "Renew License"]
+      expect(@facility_1.renew_drivers_license?(@registrant_1)).to be true
+      expect(@registrant_1.license_data).to eq({:written=>true, :license=>true, :renewed=>true})
+    end
+  end
```

Comment:
So above here are integration tests because they're testing methods that are using helper methods. We'll also want to write unit tests for those helper methods you created to ensure they work as expected in isolation.